weave/trace_server/clickhouse_trace_server_batched.py

```diff
@@ -311,7 +320,7 @@ def call_start(self, req: tsi.CallStartReq) -> tsi.CallStartRes:
             trace_id=ch_call.trace_id,
         )
 
-    def call_end(self, req: tsi.CallEndReq) -> tsi.CallEndRes:
+    def call_end(self, req: tsi.CallEndReq, publish: bool = True) -> tsi.CallEndRes:
```

Comment:
What does this `publish` control? It is very rare (if at all) that we deviate from the req/res pattern. I am curious why we need this and where it is being called from

Reply:
This is called with `publish=False` from the scoring worker. The goal is to reduce noise.
Scoring calls creates new calls (the `Scorer.score` call), which would get published to the Bufstream topic again. Of course those calls are dismissed by the worker as they do no match any monitor, but it still doubles the volume of messages received for every call to actually score.

Would you rather I add `publish` as a member of `CallEndReq`?

Reply:
no, i think this is fine, but it would be nice to explain this param in code so that we understand why it deviates from the standard external interface pattern